code/campaignclassic/src/main/java/com/adobe/marketing/mobile/FilmstripCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,510 @@
+        expandedLayout.setTextViewText(R.id.manual_carousel_filmstrip_caption, newCenterCaption);
+
+        final String messageId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.MESSAGE_ID);
+        final String deliveryId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.DELIVERY_ID);
+        final int badgeCount = intentExtras.getInt(CampaignPushConstants.IntentKeys.BADGE_COUNT);
+        final int visibility = intentExtras.getInt(CampaignPushConstants.IntentKeys.VISIBILITY);
+        final int importance = intentExtras.getInt(CampaignPushConstants.IntentKeys.IMPORTANCE);
+        final String channelId =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CHANNEL_ID);
+        final String notificationBackgroundColor =
+                intentExtras.getString(
+                        CampaignPushConstants.IntentKeys.NOTIFICATION_BACKGROUND_COLOR);
+        final String titleTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.TITLE_TEXT_COLOR);
+        final String expandedBodyTextColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT_COLOR);
+        final String smallIcon =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON);
+        final String smallIconColor =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.SMALL_ICON_COLOR);
+        final String customSound =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.CUSTOM_SOUND);
+
+        final Notification notification =
+                createNotificationBuilder(
+                                context,
+                                channelId,
+                                customSound,
+                                newCenterIndex,
+                                badgeCount,
+                                imageUrls,
+                                imageCaptions,
+                                imageClickActions,
+                                expandedLayout,
+                                smallLayout,
+                                titleText,
+                                bodyText,
+                                expandedBodyText,
+                                notificationBackgroundColor,
+                                titleTextColor,
+                                expandedBodyTextColor,
+                                messageId,
+                                deliveryId,
+                                smallIcon,
+                                smallIconColor,
+                                visibility,
+                                importance,
+                                true)
+                        .build();
+
+        notificationManager.notify(messageId.hashCode(), notification);
+    }
+
+    private static NotificationCompat.Builder createNotificationBuilder(
```

Comment:
Thinking about it, the other way to it in Java so that we are aware of the parameters is using a private builder object. Let us add a TODO to take it up once the other PR is done.